baseapp/baseapp.go

```diff
@@ -676,7 +676,11 @@ func (app *BaseApp) getContextForTx(mode execMode, txBytes []byte) sdk.Context {
 	}
 
 	if mode == execModeSimulate {
-		ctx, _ = ctx.CacheContext()
+		if ms, err := app.cms.CacheMultiStoreWithVersion(app.cms.LatestVersion()); err != nil {
```

Comment:
is this a thread safe operation?

Reply:
this is the same approach we use in serverv2, it should be safe is multistore is implemented correctly.